[1.2] Modernize buildscript, Gradle 9, GradleUtils 3 Beta

Jonathing opened the pull request.

Jonathing committed:
Modernize buildscript, Gradle 9, GradleUtils 3 Beta

Jonathing committed:
Use simplified license header for source files

Jonathing commented:
Drafted because the build is failing on `createChangelog`.

LexManos commented:
Not that i care too much, but the new header might need to mention something with it being a fork/forge now.

Jonathing committed:
Update plugins to fix createChangelog

Jonathing committed:
Generate changelog from 1.0

Jonathing committed:
Remove Groovydoc jar generation

It's completely broken on Gradle 9 due to the hard requirement on JVM
17. The Groovydoc task uses an ant task and the JVM cannot be forked.

Jonathing committed:
Fix tests failing to compile

Jonathing committed:
Bump GradleUtils and GitVersion

Jonathing committed:
Include publish task in workflow for Forge maven
